internal/testing/setup/setup.go

```diff
@@ -280,3 +281,23 @@ func NewAzureTestPipeline(ctx context.Context, t *testing.T, api string, credent
 	})
 	return p, done, httpClient
 }
+
+// FakeGCPDefaultCredentials sets up the environment with fake GCP credentials.
+// It returns a cleanup function.
+func FakeGCPDefaultCredentials(t *testing.T) func() {
+	const envVar = "GOOGLE_APPLICATION_CREDENTIALS"
+	jsonCred := []byte(`{"client_id": "foo.apps.googleusercontent.com", "client_secret": "bar", "refresh_token": "baz", "type": "authorized_user"}`)
+	f, err := ioutil.TempFile("", "fake-gcp-creds")
```

Comment:
This can be removed, `ioutil.WriteFile` will create the file if it doesn't exist.

Reply:
Right, but `TempFile` is needed to get the temporary file name.